cmd/sqlflowserver/main_test.go

```diff
@@ -1360,3 +1243,196 @@ func TestEnd2EndMaxComputePAI(t *testing.T) {
 	t.Run("CaseTrainSQL", CaseTrainSQL)
 	t.Run("CaseTrainDistributedPAI", CaseTrainDistributedPAI)
 }
+
+func TestEnd2EndWorkflow(t *testing.T) {
+	a := assert.New(t)
+	if os.Getenv("SQLFLOW_TEST_DATASOURCE") == "" || strings.ToLower(os.Getenv("SQLFLOW_TEST")) != "workflow" {
+		t.Skip("Skipping workflow test.")
+	}
+	driverName, _, err := database.ParseURL(testDatasource)
+	a.NoError(err)
+
+	if driverName != "mysql" && driverName != "maxcompute" {
+		t.Skip("Skipping workflow test.")
+	}
+	modelDir := ""
+	tmpDir, caCrt, caKey, err := generateTempCA()
+	defer os.RemoveAll(tmpDir)
+	if err != nil {
+		t.Fatalf("failed to generate CA pair %v", err)
+	}
+
+	go start(modelDir, caCrt, caKey, unitTestPort, true)
+	waitPortReady(fmt.Sprintf("localhost:%d", unitTestPort), 0)
+	if err != nil {
+		t.Fatalf("prepare test dataset failed: %v", err)
+	}
+
+	if driverName == "maxcompute" {
+		AK := os.Getenv("MAXCOMPUTE_AK")
+		SK := os.Getenv("MAXCOMPUTE_SK")
+		endpoint := os.Getenv("MAXCOMPUTE_ENDPOINT")
+		dbConnStr = fmt.Sprintf("maxcompute://%s:%s@%s", AK, SK, endpoint)
+		caseDB = os.Getenv("MAXCOMPUTE_PROJECT")
+		if caseDB == "" {
+			t.Fatalf("Must set env MAXCOMPUTE_PROJECT")
+		}
+		caseTrainTable = "sqlflow_test_iris_train"
+		caseTestTable = "sqlflow_test_iris_test"
+		casePredictTable = "sqlflow_test_iris_predict"
+		// write model to current MaxCompute project
+		caseInto = "my_dnn_model"
+
+		err = prepareTestData(dbConnStr)
+		if err != nil {
+			t.Fatalf("prepare test dataset failed: %v", err)
+		}
+	}
+
+	t.Run("CaseWorkflowTrainAndPredictDNN", CaseWorkflowTrainAndPredictDNN)
+	t.Run("CaseTrainDistributedPAIArgo", CaseTrainDistributedPAIArgo)
+}
+
+func CaseWorkflowTrainAndPredictDNN(t *testing.T) {
+	a := assert.New(t)
+	sqlProgram := fmt.Sprintf(`
+SELECT *
+FROM %s.%s
+TO TRAIN DNNClassifier
+WITH
+	model.n_classes = 3,
+	model.hidden_units = [10, 20]
+COLUMN sepal_length, sepal_width, petal_length, petal_width
+LABEL class
+INTO %s;
+
+SELECT *
+FROM %s.%s
+TO PREDICT %s.%s.class
+USING %s;
+
+SELECT *
+FROM %s.%s LIMIT 5;
+	`, caseDB, caseTrainTable, caseInto,
+		caseDB, caseTestTable, caseDB, casePredictTable, caseInto,
+		caseDB, casePredictTable)
+
+	conn, err := createRPCConn()
+	if err != nil {
+		a.Fail("Create gRPC client error: %v", err)
+	}
+	defer conn.Close()
+
+	cli := pb.NewSQLFlowClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+
+	stream, err := cli.Run(ctx, &pb.Request{Sql: sqlProgram, Session: &pb.Session{DbConnStr: testDatasource}})
+	if err != nil {
+		a.Fail("Create gRPC client error: %v", err)
+	}
+	checkWorkflow(stream)
```

Comment:
After #1512 merged, please use `Fetch` API to check the workflow works well and not only check the workflowID.